exporter/collector/metrics.go

```diff
@@ -191,13 +214,47 @@ func NewGoogleCloudMetricsExporter(
 		})
 	}
 
+	if cfg.MetricConfig.WALConfig.Enabled {
+		err = mExp.setupWAL()
+		if err != nil {
+			return nil, err
+		}
+		// start WAL popper routine
+		mExp.goroutines.Add(1)
+		go mExp.walRunner(ctx)
+	}
+
 	// Fire up the metric descriptor exporter.
 	mExp.goroutines.Add(1)
 	go mExp.exportMetricDescriptorRunner()
 
 	return mExp, nil
 }
 
+func (me *MetricsExporter) setupWAL() error {
+	walPath := filepath.Join(me.cfg.MetricConfig.WALConfig.Directory, "gcp_metrics_wal")
+	me.walPath = walPath
+	metricWal, err := wal.Open(walPath, nil)
+	if err != nil {
+		return err
+	}
+	me.wal = metricWal
+
+	// sync existing WAL indices
```

Comment:
If there are a nonzero number of entries in the WAL at startup, does it mean we're recovering from a crash/unclean shutdown? In that case it seems useful to log how many entries we're about to try to recover.

(Crash recovery seems like the biggest benefit of file- vs memory-backed WAL.)